diff --git a/tensorflow/compiler/mlir/lite/flatbuffer_export.cc b/tensorflow/compiler/mlir/lite/flatbuffer_export.cc
index 1a55e517791..b330af083a3 100644
--- a/tensorflow/compiler/mlir/lite/flatbuffer_export.cc
+++ b/tensorflow/compiler/mlir/lite/flatbuffer_export.cc
@@ -2577,10 +2577,13 @@ Translator::CreateMetadataVector() {
   }
   for (const auto& kv : metadata_) {
     const std::string& val = kv.second;
-    // Only take the first kByteStringSize values.
-    const int count = std::min(kByteStringSize, val.length());
-    std::string value = std::string(kByteStringSize, '\0')
-                            .assign(val.begin(), val.begin() + count);
+    if(val.length() % 16 != 0) {
+      module_.emitError(
+            "metadata string must be aligned to 16 bytes");
+        return std::nullopt;
+      }
+    std::string value = std::string(val.length(), '\0')
+                            .assign(val.begin(), val.begin() + val.length());
     metadata.push_back(BuildMetadata(kv.first, value));
   }
 
