--- larq_compute_engine/mlir/BUILD	2021-09-08 14:29:32.000000000 +0100
+++ BUILD2	2021-12-08 10:22:56.000000000 +0000
@@ -1,4 +1,4 @@
-load("@org_tensorflow//third_party/mlir:tblgen.bzl", "gentbl")
+load("@org_tensorflow//third_party/mlir:tblgen.bzl", "gentbl", "td_library")
 load("@org_tensorflow//tensorflow:tensorflow.bzl", "pybind_extension", "tf_cc_binary")
 
 package(
@@ -6,6 +6,15 @@
     licenses = ["notice"],  # Apache 2.0
 )
 
+td_library(
+    name = "lce_ops_td_file",
+    srcs = [
+        "ir/lce_ops.td",
+    ],
+    includes = ["include"],
+    deps = [],
+)
+
 gentbl(
     name = "lce_ops_inc_gen",
     tbl_outs = [
@@ -146,6 +155,7 @@
         "//larq_compute_engine/core:types",
         "//larq_compute_engine/core/bitpacking:bitpack",
         "@llvm-project//mlir:IR",
+        "@llvm-project//mlir:QuantOps",
     ],
 )
 
